core/src/main/scala/org/apache/spark/sql/execution/columnar/JDBCAppendableRelation.scala

```diff
@@ -276,6 +277,13 @@ case class JDBCAppendableRelation(
   def flushRowBuffer(): Unit = {
     // nothing by default
   }
+
+  override def createIndex(indexIdent: QualifiedTableName,
+                           tableIdent: QualifiedTableName,
+                           indexColumns: Seq[String],
+                           options: Map[String, String]): Unit = {
+    throw new UnsupportedOperationException("Indexes are not supported");
+  }
```

Comment:
Inconsistent indentation of functions? Use the codeStyleSettings we have been using which is as per Spark's conventions.